src/anyio/_core/_threads.py

```diff
@@ -83,29 +85,36 @@ def create_blocking_portal() -> BlockingPortal:
     return get_asynclib().BlockingPortal()
 
 
+@contextmanager
 def start_blocking_portal(
         backend: str = 'asyncio',
-        backend_options: Optional[Dict[str, Any]] = None) -> BlockingPortal:
+        backend_options: Optional[Dict[str, Any]] = None) -> Generator[BlockingPortal, Any, None]:
     """
     Start a new event loop in a new thread and run a blocking portal in its main task.
 
     The parameters are the same as for :func:`~anyio.run`.
 
     :param backend: name of the backend
     :param backend_options: backend options
-    :return: a blocking portal object
+    :return: a context manager that yields a blocking portal
 
     """
     async def run_portal():
-        nonlocal portal
-        async with create_blocking_portal() as portal:
-            event.set()
-            await portal.sleep_until_stopped()
+        async with create_blocking_portal() as portal_:
+            future.set_result(portal_)
+            await portal_.sleep_until_stopped()
 
-    portal: Optional[BlockingPortal]
-    event = threading.Event()
+    future: Future[BlockingPortal] = Future()
     kwargs = {'func': run_portal, 'backend': backend, 'backend_options': backend_options}
-    thread = threading.Thread(target=run, kwargs=kwargs)
+    thread = threading.Thread(target=run, kwargs=kwargs, daemon=True)
```

Comment:
this won't work in a sub-interpreter, and it looks like there's movement to remove the feature from the language

as @vstinner says: https://bugs.python.org/issue37266
> Daemon threads must die. That's a first step towards their death!


Per https://bugs.python.org/issue39812 using a `with concurrent.futures.ThreadPoolExecutor(max_workers=1) as pool:` should create threads that are scheduled to be cleaned up correctly

Reply:
I've switched to `ThreadPoolExecutor`, so this should not be an issue anymore.